clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -851,6 +851,33 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
                       Twine(getClangFullCPPVersion()) + "\"");
 
   // Initialize language-specific preprocessor defines.
+  if (LangOpts.getLibcxxHardeningMode()) {
+    const char *LibcxxHardeningStr;
+
+    Builder.defineMacro("_LIBCPP_HARDENING_MODE_NONE", "0");
```

Comment:
It would be great to avoid coupling these values between Clang and libc++. Can't you just define e.g. `_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_FAST` without actually defining what `_LIBCPP_HARDENING_MODE_FAST` expands to?